app.rb

```diff
@@ -19,12 +20,37 @@ class MakersBNB < Sinatra::Base
   end
 
   post '/spaces/new' do
-    Space.create(name: params[:name], description: params[:description], price: params[:price], checkin: params[:checkin], checkout: params[:checkout])
+    Space.create(
+      name: params[:name],
+      description: params[:description],
+      price: params[:price],
+      checkin: params[:checkin],
+      checkout: params[:checkout]
+    )
     redirect '/spaces'
   end
 
   get '/spaces' do
     @spaces = Space.all
     erb :'spaces/index'
   end
+
+  get '/sessions/new' do
+    erb :login
+  end
+
+  post '/sessions/new' do
+    session[:user] = User.authenticate(
+      email: params[:email],
+      password: params[:password]
+    )
+    redirect '/sessions/new' if session[:user].nil?
```

Comment:
Neat. Do these redirections tell the user why they're being redirected, or is this still a feature we're going to implement?

Reply:
That is a feature we're still yet to implement, I've added a checklist to Trello of error messages that we may want to implement going forward.